File: .claude/hooks/session-start-github-context.mjs
#!/usr/bin/env node
// @hookstack session-start-github-context
import { execSync } from "node:child_process";
// @hookstack session-start-github-context
// Injecte l'état GitHub (PRs ouvertes, checks de la branche) au démarrage de session (SessionStart)
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";

function defaultExec(cmd) {
	try {
		return execSync(cmd, {
			encoding: "utf8",
			timeout: 10_000,
			stdio: "pipe",
		}).trim();
	} catch {
		return "";
	}
}

export function run(_input, { exec = defaultExec } = {}) {
	// Silencieux si gh absent, non authentifié ou dépôt sans remote GitHub
	const prs = exec("gh pr list --state open --limit 5");
	const checks = exec("gh pr checks 2>/dev/null");

	if (!prs && !checks) return null;

	const lines = ["## GitHub Context"];
	if (prs) lines.push("### Open PRs", "```", prs, "```");
	if (checks)
		lines.push("### Checks on current branch PR", "```", checks, "```");

	return {
		hookSpecificOutput: {
			hookEventName: "SessionStart",
			additionalContext: lines.join("\n"),
		},
	};
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	const input = JSON.parse(readFileSync(0, "utf8"));
	const result = run(input);
	if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/session-start-github-context.py
#!/usr/bin/env python3
# @hookstack session-start-github-context
"""Injects GitHub state (open PRs, branch checks) at session start (SessionStart)."""
import json
import subprocess
import sys


def _exec(cmd):
    try:
        return subprocess.run(
            cmd, shell=True, capture_output=True, text=True, timeout=10, check=True
        ).stdout.strip()
    except Exception:
        return ""


def run(_input=None, *, exec_cmd=None):
    if exec_cmd is None:
        exec_cmd = _exec

    # Silencieux si gh absent, non authentifié ou dépôt sans remote GitHub
    prs = exec_cmd("gh pr list --state open --limit 5")
    checks = exec_cmd("gh pr checks 2>/dev/null")

    if not prs and not checks:
        return None

    lines = ["## GitHub Context"]
    if prs:
        lines.append("### Open PRs")
        lines.append("```")
        lines.append(prs)
        lines.append("```")
    if checks:
        lines.append("### Checks on current branch PR")
        lines.append("```")
        lines.append(checks)
        lines.append("```")

    return {
        "hookSpecificOutput": {
            "hookEventName": "SessionStart",
            "additionalContext": "\n".join(lines),
        }
    }


if __name__ == "__main__":
    data = json.load(sys.stdin) if not sys.stdin.isatty() else None
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
